include/operations/SymbolicOperation.hpp

```diff
@@ -125,6 +125,26 @@ class SymbolicOperation final : public StandardOperation {
                        [](const auto& sym) { return !sym.has_value(); });
   }
 
+  void addControls(const Controls& c) override {
+    for (auto ctrl : c) {
+      if (actsOn(ctrl.qubit)) {
+        throw QFRException(
+            "Cannot add control to operation as it already acts on "
+            "the control qubit.");
+      }
+
+      controls.insert(ctrl);
+    }
+  }
+
+  void clearControls() override { controls.clear(); }
+
+  void removeControls(const Controls& c) override {
+    for (auto ctrl : c) {
+      controls.erase(ctrl);
+    }
+  }
+
```

Comment:
```suggestion
```
I am fairly sure this can just be removed as the `SymbolicOperation` class inherits from the `StandardOperation` class and, hence, inherits all these methods directly.